saleor/graphql/account/resolvers.py

```diff
@@ -45,6 +47,19 @@ def resolve_staff_users(info, query):
     return gql_optimizer.query(qs, info)
 
 
+def resolve_user(info, id):
+    requester = info.context.user or info.context.service_account
+    if requester:
+        _model, user_pk = graphene.Node.from_global_id(id)
+        if requester.has_perms(["account.manage_staff", "account.manage_users"]):
+            return models.User.objects.filter(pk=user_pk).first()
+        if requester.has_perm("account.manage_staff"):
+            return models.User.objects.filter(pk=user_pk, is_staff=True).first()
```

Comment:
We have a `staff` queryset which you could use here. Ideally, we should try to reuse those querysets across the codebase to have one logic for getting staff users.

Reply:
It seems that `resolve_staff_users ` function above could also use that.